docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,221 @@
+
+Furthermore, if the limits are reached, how will newer data be handled? Will the pre-existing data get overwritten or will the new data be rejected? There's potential for lost data here.
+
+##### Mitigation
+
+More research needs to be done here to see what the limits are and how we can overcome them. If using RH Ingress, there could be a daily job that would export data from here and import it into our own storage.
+
+#### Risk 3: DDoS-ing Insights Ingress Server
+
+Without any constraints, it's possible for the assisted service to send data to the data collection server repeatedly. This can prevent other clients from sending their data, and cause loss of data, or prevent us from reaching the data since the server will be busy with the repeated connections.
+
+##### Mitigation
+
+1. Clearly defining when events are sent (see `Data Sending Flow` under the `Proposal` section above)
+2. Data is sent as best effort
+3. Retry will be implemented as a future enhancement
+
+#### Risk 4: Duplicate Data
+
+With all the cases for sending data, there's a high chance for duplicated data to be sent to the data collection server. It can skew the analysis of data and it won't be representative of the actual data from the field.
+
+##### Mitigation
+
+A field can be used as a marker to indicate the data has been sent to the event streaming service or that the data already exists. One option is to use the timestamp, cluster ID, and user email to detect duplicate events.
+
+## Design Details
+
+### UI Impact
+
+No UI changes will need to be made.
+
+### Test Plan
+
+#### Test 1
+
+Add a check at the end of a cluster install attempt that will query the data hosting endpoint and verify the payload made it to the data hosting service.
+
+#### Test 2
+
+Check against a large-scale cluster with > 100 hosts to verify:
+1. All of the cluster's events are being stored, queried, and sent (within a reasonable time)
+2. That the event streaming can receive and process the large data set
+
+## Drawbacks
+
+- Dependency on the data hosting service selected in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244) could cause delays in this epic being completed.
+- Won't get data from disconnected environment users since this solution requires a connection to push the data
+
+## Alternatives
+
+### Alternative Solution 1: Customer Uploads Data
+
+#### Details
+
+The Assisted Service would provide an endpoint that the customer can query to get all data that they would upload to us.
+
+#### Benefits
+
+- Support disconnected deployments since the service will not be pushing the data.
+- Minimizes risk of DDoS-ing the data streaming service
+
+#### Drawbacks
+
+- Likely will not get data
+    - Due to the customer having to do extra work to upload the data
+- Data would be skewed since we'd only get data when the customer uploads it
+    - Customer is most likely only going to upload when they have an issue
+
+---
+
+## Future Enancements
```

Comment:
I think it might be better for this to just be different epics in JIRA rather than a section in this doc.

Reply:
Thanks Nick! Removing this and creating new epics in JIRA